### PR TITLE
Removed more icons from other dialogs (bnc#875201)

### PR DESCRIPTION
**Pull request by cwh42**

### CODE REVIEW

**Comment by aschnell** [comment]
REQ: Changelog is missing. Esp. since features are removed.

File: library/wizard/src/modules/Confirm.rb [hunk 1]
```diff
@@ -55,7 +54,7 @@ def main
 
     # Confirm hardware detection (only in manual installation)
     # @param [String] class hardware class (network cards)
-    # @param [String] icon_name name of the icon. If nil, generic hardware icon will be used
+    # @param [String] obsolete - just for compatibility
     # @return true on continue
     def Detection(class_, icon_name)
```
Comment (jreidinger) [inline library/wizard/src/modules/Confirm.rb]: you use it somewhere or rubocop should complain, that unused parameter should start with _

**Comment by cwh42** [comment]
So there are two public methods whose API contain an icon-parameters. I kept those parameters to not change the API. Maybe an deprecation warning would be the right thing here but I am not sure if we do such things as deprecation warnings.
Changelog follows.

File: library/wizard/src/modules/Progress.rb [hunk 2]
```diff
@@ -677,33 +586,9 @@ def SubprogressTitle(title)
...
-      tits = deep_copy(tits)
-      icons_definition = deep_copy(icons_definition)
-      @global_visible_icons_definition = Ops.get(icons_definition, 0, [])
-      @global_invisible_icons_definition = Ops.get(icons_definition, 1, [])
-
-      @use_icons_in_progress = true
       New(window_title, progress_title, length, stg, tits, help_textmap)
-      @use_icons_in_progress = false
-
       nil
```
Comment (jreidinger) [inline library/wizard/src/modules/Progress.rb]: New already return nil, so this is not needed

**Comment by jreidinger** [comment]
travis failed on rubocop rules and I am right, you do not mark unused parameters according to convention

**Comment by lslezak** [comment]
@cwh42 to the deprecation:

- You can use `@deprecated` yard tag (see http://www.rubydoc.info/gems/yard/file/docs/Tags.md#deprecated)
- You can log the deprecation warning (e.g. via `log.warn`) with the caller location using `caller(0)` (or the old way via `Builtins.y2warning(-1, message)` to easily find where it was called from

**Comment by cwh42** [comment]
Added deprecation warnings and updated documentation.
Changelog added.

**Comment by jreidinger** [comment]
LGTM, from my side

**Comment by lslezak** [comment]
LGTM

**Pull request merged**